podman/domain/manifests.py

```diff
@@ -0,0 +1,247 @@
+
+        Args:
+            client: Connection to Podman service.
+        """
+        super().__init__(client)
+
+    def exists(self, key: str) -> bool:
+        key = urllib.parse.quote_plus(key)
+        response = self.client.get(f"/manifests/{key}/exists")
+        return response.status_code == requests.codes.no_content
+
+    def create(
+        self,
+        names: List[str],
+        images: Optional[List[Union[Image, str]]] = None,
+        all: Optional[bool] = None,  # pylint: disable=redefined-builtin
+    ) -> Manifest:
+        """Create a Manifest.
+
+        Args:
+            names: Identifiers to be added to the manifest. There must be at least one.
+            images: Images or Image identifiers to be included in the manifest.
+            all: When True, add all contents from images given.
+
+        Raises:
+            ValueError: When no names are provided.
+            NotFoundImage: When a given image does not exist.
+        """
+        if names is None or len(names) == 0:
+            raise ValueError("At least one manifest name is required.")
+
+        params = {"name": names}
+        if images is not None:
+            params["image"] = list()
+            for item in images:
+                if isinstance(item, Image):
+                    item = item.attrs["RepoTags"][0]
+                params["image"].append(item)
+
+        if all is not None:
+            params["all"] = all
+
+        response = self.client.post("/manifests/create", params=params)
+        body = response.json()
+
+        if response.status_code == requests.codes.okay:
+            manifest = self.get(body["Id"])
+            manifest.attrs["names"] = names
+            return manifest
+
+        if response.status_code == requests.codes.not_found:
+            raise ImageNotFound(body["cause"], response=response, explanation=body["message"])
+        raise APIError(body["cause"], response=response, explanation=body["message"])
+
+    def get(self, key: str) -> Manifest:
+        """Returns and manifest by name.
+
+        Args:
+            key: Manifest name for which to search
+
+        Raises:
+            NotFound: When manifest could not be found.
+            APIError: When service reports an error.
+
+        Notes:
+            To have Manifest more conform with other PodmanResource's we use the key that
+                retrieved the Manifest be its name.
+        """
+        quoted_key = urllib.parse.quote_plus(key)
+        response = self.client.get(f"/manifests/{quoted_key}/json")
+        body = response.json()
+
+        if response.status_code != requests.codes.okay:
+            if response.status_code == requests.codes.not_found:
+                raise NotFound(body["cause"], response=response, explanation=body["message"])
+            raise APIError(body["cause"], response=response, explanation=body["message"])
+
+        body["names"] = [key]
+        return self.prepare_model(attrs=body)
```

Comment:
Nit: For consistency-sake, would you consider sticking to the pattern used elsewhere:

1. If "good", return it.
2. If bad, tailor exception to fit the crime

Reply:
I have been putting the "shorter" path first, in this case the error path and  "go right" are the same length. So sure.